### PR TITLE
fix: cascade style respects order

### PR DESCRIPTION
**Pull request by bertho-zero**
Closes https://github.com/chakra-ui/chakra-ui/issues/6829

## 📝 Description

Using `Object.assign` overrides existing properties and does not respect order of precedence.

## ⛳️ Current behavior (updates)

The order of precedence is not respected.

## 🚀 New behavior

The order of precedence is respected.

## 💣 Is this a breaking change (Yes/No):

No

### CODE REVIEW

File: packages/legacy/utils/src/object.ts [hunk 1]
```diff
@@ -153,3 +153,25 @@ export const fromEntries = <T extends unknown>(entries: [string, any][]) =>
  */
 export const getCSSVar = (theme: Dict, scale: string, value: any) =>
   theme.__cssMap?.[`${scale}.${value}`]?.varRef ?? value
+
+/**
+ * Object.assign which moves properties to end of object rather than replacing them
+ * */
+export const assignAfter = (target: object, ...sources: any[]) => {
```
Comment (TylerAPfledderer) [inline packages/legacy/utils/src/object.ts:160]: I think you meant to stick this in `packages/utilities/object-utils` and not `packages/legacy/utils`? 😅

**Review by TylerAPfledderer — COMMENTED** [review]
This is awesome, @bertho-zero! I can only provide one tiny comment. 😸

**Comment by bertho-zero** [comment]
@TylerAPfledderer It's done, I moved the function to `object-utils`

File: .changeset/rich-singers-build.md [hunk 2]
```diff
@@ -2,4 +2,5 @@
 "@chakra-ui/radio": patch
 ---
 
-Update type of `value` and `defaultValue` to `string` instead of `string | number`. This reflects the internal implementation
+Update type of `value` and `defaultValue` to `string` instead of
+`string | number`. This reflects the internal implementation
```
Comment (bertho-zero) [inline .changeset/rich-singers-build.md:6]: Why this change after a pull ? :thinking:

**Comment by k-nearest-neighbor** [comment]
After an hour or so tracking down this bug, I see there is a PR that fixes it!
But it is waiting to be merged for 30+ days, and not clear when it will be deployed.

**Review by segunadebayo — APPROVED** [review]
Thanks for working on this @bertho-zero.

It aligns with the initial design I had in mind.

I'm a bit weary it might cause some breaking changes for some users, but the solution here is fixes a bug in the system.

Thanks again.

**Comment by bertho-zero** [comment]
Thanks for the merge @segunadebayo, there is another system/styled-system related issue you might be interested in https://github.com/chakra-ui/chakra-ui/issues/7136
